### PR TITLE
PSVAMB-49001 -> Adding email notification for any HTTP notification j…

### PR DESCRIPTION
**Pull request by TalDubovKaltura**
…ob that failed of some object

### CODE REVIEW

**Review by hilak — CHANGES_REQUESTED** [review]
@TalDubovKaltura this notification needs additional conditions - please add condition for objectType (ASSET), and check for $scope->getEvent()->getBatchJob()->getObjectId() to be the ID of a caption asset.

**Comment by TalDubovKaltura** [comment]
@hilak This was my previous question on the last PR as for the solution, do we want it to be specifically for caption assets and not generic? so the notification title should also describe that **caption**batchJobEmailNotificationFailure.xml

And what do you mean by _be the ID of a caption asset._ ? we can't really know from the objectId value if its caption asset or another one

**Comment by hilak** [comment]
@TalDubovKaltura  since this is a notification dispatched as a response to a batch job status event, we need to make it as specific as possible. This was the whole reason I made the change to store the object ID and batch job object type on non-entry objects in the first place.
Re: the asset ID -  AFAIK get_class(assetPeer::retrieveByPk($scope->getEvent()->getBatchJob()->getObjectId())) will return 'CaptionAsset' for a caption asset. You could do a boolean condition.

**Comment by TalDubovKaltura** [comment]
@hilak I [added](https://github.com/kaltura/server/pull/12593/commits/aede6e8a7321d5e236d2bccd0fcd22a7928735a1) the logic of "only caption asset".
BTW, the caption asset id is not primary key so I needed to use retrieveById as you can see in the code change

File: deployment/updates/scripts/xml/notifications/2024_01_18_batchJobEmailNotificationFailure.xml [hunk 1]
```diff
@@ -43,10 +43,22 @@
...
+            <field objectType="KalturaEvalBooleanField">
+              <code>(assetPeer::retrieveById($scope->getEvent()->getBatchJob()->getObjectId()))->getType() == CaptionPlugin::getAssetTypeCoreValue(CaptionAssetType::CAPTION)</code>
+            </field>
+          </item>
         </eventConditions>
         <format>1</format>
-        <subject>[Kaltura] - Http notification failed</subject>
-        <body><![CDATA[ <p>Entry with ID {entry_id} failed with HTTP notification</p> entry name:{entry_name}  - PID:{pid} - ObjectId: {object_id} <p>Message:{event_notification_msg}</p> ]]></body>
+        <subject>[Kaltura] - Http notification for captions has failed</subject>
+        <body><![CDATA[ <p>Entry with ID {entry_id} failed with HTTP notification</p> entry name:{entry_name}  - PID:{pid} - CaptionAssetId: {object_id} <p>Message:{event_notification_msg}</p> ]]></body>
```
Comment (hilak) [inline deployment/updates/scripts/xml/notifications/2024_01_18_batchJobEmailNotificationFailure.xml:61]: Replace body text to:
HTTP notification failed for PID {pid} entry ID {entry_id} entry name {entry_name} caption asset ID {object_id} <p>Message:{event_notification_msg}

File: deployment/updates/scripts/xml/notifications/2024_01_18_batchJobEmailNotificationFailure.xml [hunk 2]
```diff
@@ -43,10 +43,22 @@
...
+            <description>Object is caption asset</description>
+            <field objectType="KalturaEvalBooleanField">
+              <code>(assetPeer::retrieveById($scope->getEvent()->getBatchJob()->getObjectId()))->getType() == CaptionPlugin::getAssetTypeCoreValue(CaptionAssetType::CAPTION)</code>
+            </field>
+          </item>
         </eventConditions>
         <format>1</format>
-        <subject>[Kaltura] - Http notification failed</subject>
-        <body><![CDATA[ <p>Entry with ID {entry_id} failed with HTTP notification</p> entry name:{entry_name}  - PID:{pid} - ObjectId: {object_id} <p>Message:{event_notification_msg}</p> ]]></body>
+        <subject>[Kaltura] - Http notification for captions has failed</subject>
```
Comment (hilak) [inline deployment/updates/scripts/xml/notifications/2024_01_18_batchJobEmailNotificationFailure.xml:60]: Replace text to:
[Kaltura] - caption asset webhook has failed

**Review by hilak — CHANGES_REQUESTED** [review]

File: deployment/updates/scripts/xml/notifications/2024_01_18_batchJobEmailNotificationFailure.xml [hunk 3]
```diff
@@ -16,9 +16,9 @@
   <multirequest>
     <request service="eventNotificationTemplate" action="add" plugin="eventNotification" partnerId="0">
       <template objectType="KalturaEmailNotificationTemplate">
-        <name>HTTP notification failed</name>
+        <name>HTTP notification for captions job has failed</name>
```
[the remark this answers is not in the document]
Comment (TalDubovKaltura) [reply deployment/updates/scripts/xml/notifications/2024_01_18_batchJobEmailNotificationFailure.xml:19]: Done.

File: deployment/updates/scripts/xml/notifications/2024_01_18_batchJobEmailNotificationFailure.xml [hunk 4]
```diff
@@ -43,10 +43,22 @@
               <code>$scope->getEvent()->getBatchJob()->getStatus() == BatchJob::BATCHJOB_STATUS_FAILED</code>
             </field>
           </item>
+          <item objectType="KalturaEventFieldCondition">
+            <description>Object is asset</description>
+            <field objectType="KalturaEvalBooleanField">
+              <code>!is_null(assetPeer::retrieveById($scope->getEvent()->getBatchJob()->getObjectId()))</code>
```
[the remark this answers is not in the document]
Comment (TalDubovKaltura) [reply deployment/updates/scripts/xml/notifications/2024_01_18_batchJobEmailNotificationFailure.xml]: Good best practice! Done.

File: deployment/updates/scripts/xml/notifications/2024_01_18_batchJobEmailNotificationFailure.xml (hunk 1 above)
Comment (TalDubovKaltura) [reply deployment/updates/scripts/xml/notifications/2024_01_18_batchJobEmailNotificationFailure.xml]: Done.

File: deployment/updates/scripts/xml/notifications/2024_01_18_batchJobEmailNotificationFailure.xml (hunk 2 above)
Comment (TalDubovKaltura) [reply deployment/updates/scripts/xml/notifications/2024_01_18_batchJobEmailNotificationFailure.xml]: Done.

**Review by hilak — APPROVED** [review]

**Pull request merged**